apps/challenges/views.py

```diff
@@ -1740,14 +1740,88 @@ def get_all_submissions_of_challenge(
         return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
 
 
+@swagger_auto_schema(
+    methods=["get"],
+    manual_parameters=[
+        openapi.Parameter(
+            name="challenge_pk",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_NUMBER,
+            description="Challenge ID",
+            required=True,
+        ),
+        openapi.Parameter(
+            name="challenge_phase_pk",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_NUMBER,
+            description="Challenge Phase ID",
+            required=True,
+        ),
+        openapi.Parameter(
+            name="file_type",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_STRING,
+            description="File Type",
+            required=True,
+        ),
+    ],
+    operation_id="download_all_submissions",
+    responses={
+        status.HTTP_200_OK: openapi.Response(""),
+        status.HTTP_400_BAD_REQUEST: openapi.Response(
+            "{'error': 'The file type requested is not valid!'}"
+        ),
+        status.HTTP_404_NOT_FOUND: openapi.Response(
+            "{'error': 'Challenge Phase does not exist'}"
+        )
+    },
+)
+@swagger_auto_schema(
+    methods=["post"],
+    manual_parameters=[
+        openapi.Parameter(
+            name="challenge_pk",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_NUMBER,
+            description="Challenge ID",
+            required=True,
+        ),
+        openapi.Parameter(
+            name="challenge_phase_pk",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_NUMBER,
+            description="Challenge Phase ID",
+            required=True,
+        ),
+        openapi.Parameter(
+            name="file_type",
+            in_=openapi.IN_PATH,
+            type=openapi.TYPE_STRING,
+            description="File Type",
+            required=True,
+        ),
+    ],
+    operation_id="download_all_submissions",
+    responses={
+        status.HTTP_200_OK: openapi.Response(""),
+        status.HTTP_400_BAD_REQUEST: openapi.Response(
+            "{'error': 'The file type requested is not valid!'}"
+        ),
+        status.HTTP_401_UNAUTHORIZED: openapi.Response(
+            "{'error': 'Sorry, you do not belong to this Host Team!'}"
+        )
+    },
+)
 @api_view(["GET", "POST"])
 @throttle_classes([UserRateThrottle])
 @permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
 @authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
 def download_all_submissions(
     request, challenge_pk, challenge_phase_pk, file_type
 ):
-
+    """
+    API endpoint to download all the submissions for a particular challenge as a csv
```

Comment:
@afif-fahim please complete the docstrings i.e mention Arguments and API response